doozer/doozerlib/cli/release_gen_payload.py

```diff
@@ -761,25 +761,44 @@ async def sync_payloads(self):
             False: dict()
         }
 
+        # Filter out embargoed builds to mirror to public quay
+        filtered_payload_entries_for_arch = {}
+        for arch, payload_entries in self.payload_entries_for_arch.items():
+            filtered_payload_entries = {}
+            for payload_tag_name, payload_entry in payload_entries.items():
+                if payload_entry.build_inspector.is_under_embargo():
+                    # We do not want embargoed builds to get leaked to public quay
+                    continue
+                filtered_payload_entries[payload_tag_name] = payload_entry
+
+            if filtered_payload_entries:
+                # Sometimes all builds might be embargoed, hence the variable can be empty
+                filtered_payload_entries_for_arch[arch] = filtered_payload_entries
+
         # Ensure that all payload images have been mirrored before updating
         # the imagestream. Otherwise, the imagestream will fail to import the
         # image.
         tasks = []
         for arch, payload_entries in self.private_payload_entries_for_arch.items():
             tasks.append(self.mirror_payload_content(arch, payload_entries, True))
-        for arch, payload_entries in self.payload_entries_for_arch.items():
+        for arch, payload_entries in filtered_payload_entries_for_arch.items():
             tasks.append(self.mirror_payload_content(arch, payload_entries))
         await asyncio.gather(*tasks)
 
         await asyncio.sleep(120)
 
-        # Update the imagestreams being monitored by the release controller.
-        tasks = []
-        for arch, payload_entries in self.payload_entries_for_arch.items():
-            for private_mode in self.privacy_modes:
+        # Updating public and private image streams
+        for private_mode, payload_entries_for_each_arch in [(False, filtered_payload_entries_for_arch),
+                                                            (True, self.private_payload_entries_for_arch)]:
+            tasks = []
+            for arch, payload_entries in payload_entries_for_each_arch:
                 self.logger.info(f"Building payload files for architecture: {arch}; private: {private_mode}")
-                tasks.append(self.generate_specific_payload_imagestreams(arch, private_mode, payload_entries, multi_specs))
-        await asyncio.gather(*tasks)
+                tasks.append(self.generate_specific_payload_imagestreams(arch,
+                                                                         private_mode,
+                                                                         payload_entries,
+                                                                         multi_specs
+                                                                         ))
```

Comment:
when it comes to parameter list indentation, i prefer pretty much any other way that pep8 allows. why? well, not only does this take up more lines and leave your content way out right where you don't see it, but also it needs to be re-indented if you ever refactor names or change something else about the first line.

so i'd suggest:
```suggestion
                tasks.append(self.generate_specific_payload_imagestreams(
                    arch, private_mode, payload_entries, multi_specs))
```